Source/SwiftLintFramework/Extensions/Configuration+Parsing.swift

```diff
@@ -194,14 +199,43 @@ extension Configuration {
                 }
 
             case let .default(disabled: disabledRules, optIn: optInRules):
-                if rule is OptInRule.Type, Set(optInRules).isDisjoint(with: rule.description.allIdentifiers) {
-                    queuedPrintError("\(message), but it is not enabled on " +
-                        "'\(Key.optInRules.rawValue)'.")
-                } else if Set(disabledRules).isSuperset(of: rule.description.allIdentifiers) {
-                    queuedPrintError("\(message), but it is disabled on " +
-                        "'\(Key.disabledRules.rawValue)'.")
+                validateConfiguredRulesAreEnabled(
+                    message: message,
+                    parentConfiguration: parentConfiguration,
+                    disabledRules: disabledRules,
+                    optInRules: optInRules,
+                    rule: rule
+                )
+            }
+        }
+    }
+
+    private static func validateConfiguredRulesAreEnabled(
+        message: String,
+        parentConfiguration: Configuration?,
+        disabledRules: Set<String>,
+        optInRules: Set<String>,
+        rule: Rule.Type
+    ) {
+        if rule is OptInRule.Type {
+            var allOptInRules = optInRules
+            if let parentConfiguration {
+                switch parentConfiguration.rulesMode {
+                case .allEnabled:
+                    return
+                case .only(let parentOnlyRules):
+                    allOptInRules.formUnion(parentOnlyRules)
```

Comment:
Same here: Don't we have to subtract the rules disabled in the child configuration?